pkg/utils/datamigrate.go

```diff
@@ -52,64 +52,78 @@ func GetDataMigrateJobName(releaseName string) string {
+		err = fmt.Errorf("invalid spec: either %v or %v must be set", field.NewPath("spec").Child("to").Child("dataset"), field.NewPath("spec").Child("from").Child("dataset"))
+		return
+	}
 
-		// if runtimeType is not specified, we will use the toDataset as the targetDataset
-		if dataMigrate.Spec.RuntimeType == "" {
-			targetDataset = toDataset
-			return
+	// if runtimeType is not specified, we will simply use the toDataset as the first choice, and the fromDataset as the second.
+	if dataMigrate.Spec.RuntimeType == "" {
+		var targetDataset *datav1alpha1.DatasetToMigrate
+		if dataMigrate.Spec.To.DataSet != nil && len(dataMigrate.Spec.To.DataSet.Name) > 0 {
+			targetDataset = dataMigrate.Spec.To.DataSet
+		} else {
+			targetDataset = dataMigrate.Spec.From.DataSet
 		}
 
-		// if runtimeType is specified, check if toDataset's accelerate runtime type is the same as the runtimeType
-		index, boundedRuntime := GetRuntimeByCategory(toDataset.Status.Runtimes, common.AccelerateCategory)
-		if index == -1 {
-			err = fmt.Errorf("bounded accelerate runtime not ready")
-			return
+		var namespace string = targetDataset.Namespace
+		if len(namespace) == 0 {
+			namespace = dataMigrate.Namespace
 		}
-		if boundedRuntime.Type == dataMigrate.Spec.RuntimeType {
-			targetDataset = toDataset
-			return
+		namespacedName = types.NamespacedName{
+			Namespace: namespace,
+			Name:      targetDataset.Name,
 		}
-		boundedRuntimeType = boundedRuntime.Type
+		return
 	}
-	if dataMigrate.Spec.From.DataSet != nil && dataMigrate.Spec.From.DataSet.Name != "" {
-		fromDataset, err = GetDataset(client, dataMigrate.Spec.From.DataSet.Name, dataMigrate.Spec.From.DataSet.Namespace)
-		if err != nil {
-			return
-		}
-		// if runtimeType is not specified, we will use the fromDataset as the targetDataset
-		if dataMigrate.Spec.RuntimeType == "" {
-			targetDataset = fromDataset
-			return
-		}
 
-		// if runtimeType is specified, check if fromDataset's accelerate runtime type is the same as the runtimeType
-		index, boundedRuntime := GetRuntimeByCategory(fromDataset.Status.Runtimes, common.AccelerateCategory)
-		if index == -1 {
-			err = fmt.Errorf("bounded accelerate runtime not ready")
-			return
+	// when runtimeType is explicitly set, check whether toDataset or fromDataset has the runtimeType
+	datasetsToCheck := []*datav1alpha1.DatasetToMigrate{dataMigrate.Spec.To.DataSet, dataMigrate.Spec.From.DataSet}
+	checkedRuntimeTypes := []string{}
+	for _, toCheck := range datasetsToCheck {
+		if toCheck != nil && len(toCheck.Name) > 0 {
+			var namespace string = toCheck.Namespace
+			if len(namespace) == 0 {
+				namespace = dataMigrate.Namespace
+			}
+
+			dataset, innerErr := GetDataset(client, toCheck.Name, namespace)
+			if innerErr != nil {
+				err = errors.Wrapf(innerErr, "failed to get dataset \"%s/%s\" from DataMigrate \"%s/%s\"'s spec", namespace, toCheck.Name, dataMigrate.Namespace, dataMigrate.Name)
```

Comment:
we can not wrap the error, as the code in 
https://github.com/fluid-cloudnative/fluid/blob/d23bf2c0050a070f4035762bd5122f19fa9098c3/pkg/controllers/operation_controller.go#L88
calls 
https://github.com/fluid-cloudnative/fluid/blob/d23bf2c0050a070f4035762bd5122f19fa9098c3/pkg/ddc/base/operation_lock.go#L92
will check the not found error.

Reply:
@xliuqq Thank you! Very careful advice. I've tested such case and I think that would not be a problem. 

In `vendor/k8s.io/apimachinery/pkg/api/errors/errors.go`, you can see the function automatically unwrap the error and check if it's a NotFound error.
```golang
// IsNotFound returns true if the specified error was created by NewNotFound.
// It supports wrapped errors and returns false when the error is nil.
func IsNotFound(err error) bool {
	reason, code := reasonAndCodeForError(err)
	if reason == metav1.StatusReasonNotFound {
		return true
	}
	if _, ok := knownReasons[reason]; !ok && code == http.StatusNotFound {
		return true
	}
	return false
}
```

I've also tried something like this, and it prints "Wrapped error supported". From my understanding, the code now should works fine.
```golang
func main() {
	err := apierrors.NewNotFound(schema.ParseGroupResource("datasets.data.fluid.io"), "dataset not found")
	wrappedErr := errors.Wrap(err, "A wrapped error")

	if apierrors.IsNotFound(wrappedErr) {
		fmt.Println("Wrapped error supported")
	} else {
		fmt.Println("Wrapped error is not support")
	}
}
```